test/test_oversized_bag.cpp

```diff
@@ -0,0 +1,222 @@
+    YGM_ASSERT_RELEASE(bbag.count("red") == 1);
+    YGM_ASSERT_RELEASE(bbag.size() == 3);
+    bbag.clear();
+  }
+
+  //
+  // Test all ranks async_insert
+  {
+    ygm::container::oversized_bag<std::string> bbag(world);
+    bbag.async_insert("dog");
+    bbag.async_insert("apple");
+    bbag.async_insert("red");
+    YGM_ASSERT_RELEASE(bbag.size() == 3 * (size_t)world.size());
+    YGM_ASSERT_RELEASE(bbag.count("dog") == (size_t)world.size());
+    YGM_ASSERT_RELEASE(bbag.count("apple") == (size_t)world.size());
+    YGM_ASSERT_RELEASE(bbag.count("red") == (size_t)world.size());
+
+    {
+      std::vector<std::string> all_data;
+      bbag.gather(all_data, 0);
+      if (world.rank0()) {
+        YGM_ASSERT_RELEASE(all_data.size() == 3 * (size_t)world.size());
+      }
+    }
+    {
+      std::set<std::string> all_data;
+      bbag.gather(all_data, 0);
+      if (world.rank0()) {
+        YGM_ASSERT_RELEASE(all_data.size() == 3);
+      }
+    }
+  }
+
+  //
+  // Test Rebalance
+  {
+    ygm::container::oversized_bag<std::string> bbag(world);
+    bbag.async_insert("begin", 0);
+    bbag.async_insert("end", world.size() - 1);
+    bbag.rebalance();
+    YGM_ASSERT_RELEASE(bbag.local_size() == 2);
+  }
+
+  // Currently, shuffle functions are being ignored as per Roger
+  // Test local_shuffle and global_shuffle
+  // {
+  //   ygm::container::oversized_bag<int> bbag(world);
+  // 
+  // }
+
+  // 
+  // Test for_all
+  {
+    ygm::container::oversized_bag<std::string> bbag(world);
+    if (world.rank0()) {
+      bbag.async_insert("dog");
+      bbag.async_insert("apple");
+      bbag.async_insert("red");
+    }
+    int count{0};
+    bbag.for_all([&count](std::string& mstr) { ++count; });
+    int global_count = world.all_reduce_sum(count);
+    world.barrier();
+    YGM_ASSERT_RELEASE(global_count == 3);
+  }
+
+  //
+  // Test for_all (pair)
+  {
+    ygm::container::oversized_bag<std::pair<std::string, int>> pbag(world);
+    if (world.rank0()) {
+      pbag.async_insert({"dog", 1});
+      pbag.async_insert({"apple", 2});
+      pbag.async_insert({"red", 3});
+    }
+    int count{0};
+    pbag.for_all(
+      [&count] (std::pair<std::string, int>& mstr) { count += mstr.second; });
+    int global_count = world.all_reduce_sum(count);
+    world.barrier();
+    YGM_ASSERT_RELEASE(global_count == 6);
+  }
+
+  //
+  // Test for_all (split pair)
+  // {
+  //   ygm::container::oversized_bag<std::pair<std::string, int>> pbag(world);
+  //   if (world.rank0()) {
+  //     pbag.async_insert({"dog"}, 1);
+  //     pbag.async_insert({"apple"}, 2);
+  //     pbag.async_insert({"red"}, 3);
+  //   }
+  //   int count{0};
+  //   pbag.for_all(
+  //     [&count](std::string& first, int& second) { count += second; });
+  //   int global_count = world.all_reduce_sum(count);
+  //   world.barrier();
+  //   YGM_ASSERT_RELEASE(global_count == 6);
+  // }
+
+
+  // Test rebalance
+  {
+    ygm::container::oversized_bag<std::string> bbag(world);
```

Comment:
needs a bbag.clear() this looks like a copy from line 102, is that intended?

Reply:
This was not intended, something that slipped through the cracks when merging, ty for pointing it out!

Reply:
good catch